### PR TITLE
Nw4--JavScript-Core-1-Coursework-Week-3-Bimbola Lasisi

### PR DESCRIPTION
**Pull request by blasisi**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered 3-extra/1-card-vailidator.md](https://github.com/blasisi/JavaScript-Core-1-Coursework-Week3/blob/Jsweek3-core1/3-extra/1-card-vailidator.md)

### CODE REVIEW

File: 2-mandatory/2-oxygen-levels.js [hunk 1]
```diff
@@ -10,8 +10,36 @@
 
     Some string methods that might help you here are .replace() and .substring(). 
 */
+// const findSafeOxygenLevel = ["24.2%", "11.3%", "19.9%", "23.1%", "29.3%", "20.2%"];
+
+// input is  string array
+//  output is returning string 
+
+ function safeLevels(arr){
```
Comment (nax2uk) [inline 2-mandatory/2-oxygen-levels.js:18]:
```suggestion
function findSafeOxygenLevel(arr) {
```

File: 2-mandatory/2-oxygen-levels.js [hunk 2]
```diff
@@ -10,8 +10,36 @@
...
+   const convert = arr.map(function(element){
+      return parseFloat(element);
+   });
```
Comment (nax2uk) [inline 2-mandatory/2-oxygen-levels.js:21]:
```suggestion
   // filter out elements that does not have a '%' symbol at the back first.
   const percentages = arr.filter(element=>element.endsWith('%'));
   const convert = percentages.map(function(element){
      // remove the '%' symbol before you parse the string as a float
      return parseFloat(element.replace('%',''));
   })
```

File: 2-mandatory/2-oxygen-levels.js [hunk 3]
```diff
@@ -10,8 +10,36 @@
...
+     const safeLevels = convert.filter(function(element){
+       return element > 19.5 && element < 23.5;
+     });
+      return `${safeLevels[0]}%`;
```
Comment (nax2uk) [inline 2-mandatory/2-oxygen-levels.js:25]:
```suggestion
      // saveLevels might have the value undefined
      if (safeLevels) return `${safeLevels}%`;
```

File: 2-mandatory/2-oxygen-levels.js [hunk 4]
```diff
@@ -10,8 +10,36 @@
...
+// const findSafeOxygenLevel = ["24.2%", "11.3%", "19.9%", "23.1%", "29.3%", "20.2%"];
+
+// input is  string array
+//  output is returning string 
+
+ function safeLevels(arr){
+   const convert = arr.map(function(element){
+      return parseFloat(element);
+   });
+     const safeLevels = convert.filter(function(element){
```
Comment (nax2uk) [inline 2-mandatory/2-oxygen-levels.js:22]:
```suggestion
     // use `.find` to return the first element found
     const safeLevels = convert.find(function(element){
```

File: 2-mandatory/5-eligible-students.js [hunk 5]
```diff
@@ -7,7 +7,12 @@
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
 */
 
-function getEligibleStudents() {}
+function getEligibleStudents(arr) {
+  arr.map(elem[1]) => {
+  return arr.filter(Element => element[1] >= 8) && element.map("attendance") >= element[0]));
```
Comment (nax2uk) [inline 2-mandatory/5-eligible-students.js:12]: really close!!!! 😍

```suggestion
  return arr.filter(element => element[1] >= 8).map(element => element[0]);
```

File: 2-mandatory/6-journey-planner.js [hunk 6]
```diff
@@ -20,8 +20,8 @@
 function checkCodeIsThere(stringText) {
   let magicWord = "code";
   //edit code below
-  if (stringText) {
-    return stringText;
+  if (stringText.includes(magicWord)) {
+    return stringText.includes(magicWord);
```
Comment (nax2uk) [inline 2-mandatory/6-journey-planner.js:24]:
```suggestion
    return stringText.indexOf(magicWord);
```

File: 2-mandatory/8-password-validator.js [hunk 7]
```diff
@@ -24,7 +24,39 @@ PasswordValidationResult=  [false, false, false, false, true]
...
+  // const character = passwords.length < 5;
+  // const number = passwords.match(/0-9]/);
+  // const lowerCase = passwords(/a-z]/);
+  
+  // const upperCase = passwords(/A-Z]/);
+  // const symbol = passwords /[/!#$%.*&]/
+  // const matchPassword = passwords.map((character, number,lowerCase,upperCase,symbol))=>{
+  // return character &&number && lowerCase && upperCase && symbol ?true : false;
+
+
```
Comment (nax2uk) [inline 2-mandatory/8-password-validator.js:59]:
```suggestion
function validatePasswords(passwords) {
  return passwords.map((password, index, passwords) => {
    const patterns = [/.{5,}/, /[A-Z]/, /[a-z]/, /[0-9]/, /[!#$%.*&]/];
    if (!patterns.every((pattern) => pattern.test(password))) {
      return false;
    }
    return index === passwords.indexOf(password);
  });
}
```

**Review by nax2uk — COMMENTED** [review]
Hi @blasisi,

This is very well done!!! Super!! 🥳

Please see my feedback .

File: 2-mandatory/2-oxygen-levels.js (hunk 3 above)
Comment (ahmd202) [reply 2-mandatory/2-oxygen-levels.js:25]: good way in using filter method ,well Done bimbola

**Pull request closed without merging**